src/components/Footer.tsx

```diff
@@ -0,0 +1,59 @@
+import cn from 'classnames';
+import { useContext, useEffect, useState } from 'react';
+import { DispatchContext, StateContext } from '../Store';
+import { Filter } from '../types/Filter';
+
+export const Footer = () => {
+  const { todos, filter } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+  const [itemsLeft, setItemsLeft] = useState(0);
+  const [hasCompletedTodos, setHasCompletedTodos] = useState(false);
+
+  useEffect(() => {
+    const remainingItems = todos.filter(todo => !todo.completed).length;
+    const isTodoCompleted = todos.some(todo => todo.completed);
+
+    setItemsLeft(remainingItems);
+    setHasCompletedTodos(isTodoCompleted);
```

Comment:
we don't need a separate state for these values

You can create const variables (but you can wrap them into useMemo) for them like you did above (13 - 14 lines) but directly in the component instead of useEffect. The reason - they will be recomputed automatically everytime when context value changes because it'll trigger this component rerender

General rule - do not create state for value if you can compute this value based on other state/props/context value